gazebo_plugins/include/gazebo_plugins/gazebo_ros_template.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GAZEBO_PLUGINS__GAZEBO_ROS_TEMPLATE_HPP_
+#define GAZEBO_PLUGINS__GAZEBO_ROS_TEMPLATE_HPP_
+
+#include <gazebo/common/Plugin.hh>
+
+// For std::unique_ptr, could be removed
+#include <memory>
+
+namespace gazebo
```

Comment:
Not sure if we should put all of these in the gazebo namespace like in ROS1. Perhaps put it in the gazebo_plugins namespace and declare ```using namespace gazebo```

Reply:
Yeah I had the same thought, but since I couldn't think of a strong reason to change it, I just left the same to reduce the amount of things changing. One reason to keep it would be to follow the same pattern as the plugins [shipped with Gazebo](https://bitbucket.org/osrf/gazebo/src/01c7f8b1d68448bc618b575ad1c7ec13fee2b87f/plugins/?at=default), but there's little meaning if that's the only pattern they share... I'll change it to `gazebo_plugins` and we can revisit this in case someone complains.

Reply:
done in https://github.com/ros-simulation/gazebo_ros_pkgs/pull/783/commits/2bc125b356bcaa39aef6c139c48e697f1f701c5e